feat(web): scenario-list, result-list API Development

kobenlys opened the pull request:
<!--  
Thank you for contributing to the API GHOST project.  
This document provides guidelines to ensure smooth collaboration and maintain high code quality.  
-->

<!--
PR Title Format Guideline

Use the following format for your PR title:

    type(scope): concise description

Examples:
    feat(api): add scenario execution support
    fix(parser): resolve YAML parsing error
    docs(readme): update CLI usage instructions

Available types:
    feat:     Add a new feature
    fix:      Fix a bug
    build:    Modify build system or external dependencies
    chore:    Modify configuration files unrelated to source or test files (e.g., .gitignore, .editorconfig)
    ci:       Update CI configuration files and scripts
    test:     Add or update tests
    docs:     Update documentation (e.g., README)
    refactor: Code changes that neither fix a bug nor add a feature
    style:    modify CSS styles
-->

### Related Issue

- #14 

### Description

- Based on API Apecification, implementation to /apighost/scenario-list, /apighost/result-list
- The folder structure for apighost has been set up within the library.

### Testing

#### 1. apighost(lib) - default folder structure  
<img src="https://github.com/user-attachments/assets/72e25396-9271-4a1f-b5b6-202bfb5e2ec7" alt="apighost folder structure" style="max-width: 60%; height: auto; display: block; margin: 1em auto;">

#### 2. `/apighost/scenario-list`  
<img src="https://github.com/user-attachments/assets/3912d2fd-335c-4be5-9b0a-6d8ffec79030" alt="scenario list" style="max-width: 60%; height: auto; display: block; margin: 1em auto;">

#### 3. `/apighost/result-list`  
<img src="https://github.com/user-attachments/assets/9f624660-16b8-4fb9-84dc-0c2d117a5923" alt="result list" style="max-width: 60%; height: auto; display: block; margin: 1em auto;">

### Additional Notes

### Pre-Submission Checklist

- [x] Have you completed code style (convention) checks locally?
- [x] Have you passed the CI tests in your local environment?

Commit:
chore: import parser module

Commit:
chore: add default path and format

Clear renaming to avoid duplication with this project

Commit:
refactor: changed directory

executor to model

Commit:
feat: Create been registration and default-setting class

Commit:
refactor: Reflections based on model module changes

Commit:
feat: scenario-list, result-list API Development

Based on API specification, implemented an API to get a list of scenarios and results

cod0216 commented on web/src/main/java/com/apighost/agent/executor/ScenarioTestExecutor.java:
```diff
@@ -96,7 +97,7 @@ public void testExecutor(String scenarioName, SseEmitter emitter) throws IOExcep
                     .endTime(responseResult.getEndTime().toString())
                     .durationMs(responseResult.getDurationMs())
-                    .requestSuccess(matchedThen != null)
+                    .isRequestSuccess(matchedThen != null)
```
naming is good!

kobenlys replied:
it not mine...

cod0216 commented on web/src/main/java/com/apighost/agent/loader/FileLoader.java:
```diff
@@ -0,0 +1,60 @@
+        return scenarioNames;
+    }
+
+    public List<File> getScenarioResults() {
```
I think `getScenarioResults()` and `getScenarioNames()` have a lot of duplicated code, so the common logic could be extracted into a separate method.

kobenlys replied:
OH NICE PERSPECTIVE! @cod0216 
GOAT ON GITHUB!!!

cod0216 commented:
to. @kobenlys 
everything else looks fantastic!
Thank you so much for working hard even late into the night — I really appreciate it!👍👍👍

kobenlys commented:
> to. @kobenlys everything else looks fantastic! Thank you so much for working hard even late into the night — I really appreciate it!👍👍👍

Im Happy~ Thank you @cod0216 !

oneweeeek commented on web/src/main/java/com/apighost/agent/engine/FileLoaderEngine.java:
```diff
@@ -0,0 +1,47 @@
+        return new ScenarioListResponse(fileLoader.getScenarioNames());
+    }
+
+    public ScenarioResultListResponse getScenarioResults() {
```
Thank you for your efforts.😊If fileLoaderEngine fails in objectMapper.readValue(), it ignores the entire result and throws an exception, so why not log the failed file and continue processing it?? @kobenlys

kobenlys replied:
Appreciated the feedback 👍👍
Sounds like a good direction. I’ll go ahead and apply the change right away!

kobenlys committed:
feat: improve of fault tolerance

The implementation has been updated to ensure that a failure during JSON parsing does not interrupt the overall processing flow.

haerim-kweon commented on web/src/main/java/com/apighost/agent/loader/FileLoader.java:
```diff
@@ -0,0 +1,60 @@
+    public List<String> getScenarioNames() {
+        List<String> scenarioNames = new ArrayList<>();
+
+        File dir = new File(apiGhostSetting.getScenarioPath());
```
There is no problem with the logic, but it would be better if the variable names were more specific

kobenlys replied:
Thank you for the feedback!
I'll reflect it right away!

kobenlys replied:
relfected that!

haerim-kweon commented on web/src/main/java/com/apighost/agent/loader/FileLoader.java:
```diff
@@ -0,0 +1,60 @@
+        }
+
+        for (File file : files) {
+            if (file.isFile() && (file.getName().endsWith(apiGhostSetting.getFormatYaml())
```
I think it would be a good idea to separate the methods for readability.

kobenlys replied:
Are you referring to the conditions within the if statement?

kobenlys replied:
i would be refactor this code thank you!

haerim-kweon commented on web/src/main/java/com/apighost/agent/model/ScenarioListResponse.java:
```diff
@@ -0,0 +1,19 @@
+
+import java.util.List;
+
+public class ScenarioListResponse {
```
I only have a List in the field, but I wonder if separating the class was done with extensibility in mind.

kobenlys replied:
This implementation is based on the API specifications. Rather than focusing on scalability, the intention was to explicitly reflect the exact response values.

haerim-kweon reviewed (commented):
Thank you for your hard work until late at night.

kobenlys committed:
refactor: Directory variables renamed for readability improvement
